sdk/nodejs/resource.ts

```diff
@@ -174,6 +233,27 @@ export abstract class Resource {
         }
         this.__protect = !!opts.protect;
 
+        // Collapse any `Alias`es down to URNs. We have to wait until this point to do so because we do not know the
+        // default `name` and `type` to apply until we are inside the resource constructor.
+        this.__aliases = [];
+        if (opts.aliases) {
+            for (const alias of opts.aliases) {
+                this.__aliases.push(output(alias).apply(a => {
+                    if (typeof a === "string") {
+                        return output(a);
+                    } else {
+                        return createUrn(
+                            output(a.name).apply(n => n || name),
+                            output(a.type).apply(ty => ty || t),
+                            a.parent || opts.parent,
+                            a.project,
+                            a.stack,
+                        );
+                    }
+                }));
+            }
+        }
+
         if (opts.id) {
             // If this resource already exists, read its state rather than registering it anew.
             if (!custom) {
```

Comment:
note: i can see you passing 'opts' along below.  I think i'd frankly be more of a fan of generating a new frsh object with all the data we want copied along (i.e. with `{ ...opts }`) and then overwriting the two props we produce new values for (`protect` and `aliases`).

Reply:
Done.